{
 "metadata": {
  "name": "",
  "signature": "sha256:95baa7c5228edc74bd48a607b3c31482ed9de812c0204375fc71bb4b5026cf22"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import os\n",
      "os.chdir(\"./final_fastas_bp/\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "Open the RASTtk app\n",
      "\n",
      "#in terminal:\n",
      "\n",
      "cd ~/Desktop/ViralFate/PhageGenomeAssembly/final_fastas_bp/\n",
      "kbase-login juliambrown"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "allfastqs=open(\"../allfastq.txt\")   #set up name dict of original names:\n",
      "name_dict={}            \n",
      "for lin in allfastqs.readlines():\n",
      "    phage=lin.split(\"_\")[0]\n",
      "    name=lin.split(\".f\")[0]\n",
      "    name_dict[phage]=name"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "finals=[]\n",
      "for x in name_dict.keys():\n",
      "    filename=x+\"final.fasta\"\n",
      "    if (os.path.exists(filename)):\n",
      "        finals.append(x)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "\n",
      "\n",
      "#os.mkdir(\"./ToSubmit/\")\n",
      "out=open(\"annote_step1.sh\",\"w\")\n",
      "for x in finals:\n",
      "    phagename=name_dict[x]\n",
      "    filename=x+\"final.fasta\"\n",
      "    \n",
      "    out.write('rast-create-genome --scientific-name \"Vibriophage_'+phagename+'\" --genetic-code 11 --domain Virus --contigs '+filename+' > ./ToSubmit/'+x+'gto'+\"\\n\\nsleep 5\\n\\n\")\n",
      "    print(phagename)\n",
      "    \n",
      "out.close()\n",
      "!open annote_step1.sh -a textedit"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1.206.O._10N.222.51.B10\n",
        "1.237.B._10N.261.52.C5\n",
        "1.267.O._10N.286.54.A1\n",
        "1.015.O._10N.222.51.E5\n",
        "1.100.O._10N.261.45.C3\n",
        "1.239.O._10N.261.52.F6\n",
        "1.196.O._10N.286.54.E12\n",
        "1.134.O._10N.222.52.B8\n",
        "1.278.O._10N.286.54.E8\n",
        "1.215.B._10N.222.54.F7\n",
        "1.289.A._10N.286.55.E8\n",
        "1.293.O._10N.261.52.E4\n",
        "1.199.B._10N.286.55.C10\n",
        "1.087.A._10N.261.45.F7\n",
        "1.217.O._10N.261.45.A1\n",
        "1.223.O._10N.261.48.A9\n",
        "1.108.O._10N.222.51.A4\n",
        "1.259.O._10N.286.48.F4\n",
        "1.115.B._10N.222.49.B11\n",
        "2.095.B._10N.286.46.E10\n",
        "1.262.O._10N.286.51.A9\n",
        "1.155.O._10N.222.55.B3\n",
        "1.195.O._10N.286.54.C8\n",
        "1.277.A._10N.286.54.E7\n",
        "1.017.O._10N.286.55.C11\n",
        "1.104.O._10N.286.49.A12\n",
        "1.166.O._10N.261.51.C7\n",
        "1.170.O._10N.261.52.C3\n",
        "1.238.A._10N.261.52.F10\n",
        "1.275.O._10N.286.54.E11\n",
        "1.133.O._10N.222.51.E4\n",
        "1.185.O._10N.286.49.C2\n",
        "2.092.O._10N.286.52.B7\n",
        "1.030.O._10N.222.55.F9\n",
        "1.247.B._10N.261.54.E12\n",
        "1.003.O._10N.286.48.A2\n",
        "1.165.O._10N.261.51.B7\n",
        "1.271.B._10N.286.54.B4\n",
        "1.219.O._10N.261.45.E2\n",
        "1.191.O._10N.286.52.B4\n",
        "1.011.O._10N.286.49.B11\n",
        "1.118.B._10N.261.49.F6\n",
        "1.101.O._10N.261.45.C6\n",
        "1.047.O._10N.286.55.F2\n",
        "1.161.O._10N.261.48.C5\n",
        "1.286.O._10N.286.55.C4\n",
        "1.164.O._10N.261.51.A7\n",
        "1.021.A._10N.222.51.F9\n",
        "1.029.O._10N.261.55.A7\n",
        "1.076.O._10N.286.51.B7\n",
        "1.200.O._10N.286.55.E1\n",
        "1.152.O._10N.222.46.E1\n",
        "1.281.O._10N.286.54.F7\n",
        "1.012.O._10N.261.48.C12\n",
        "1.147.O._10N.286.49.E9\n",
        "1.261.O._10N.286.51.A7\n",
        "1.284.A._10N.286.55.A5\n",
        "1.287.O._10N.286.55.C7\n",
        "1.063.O._10N.261.45.C7\n",
        "1.265.O._10N.286.52.F6\n",
        "1.004.O._10N.261.54.A2\n",
        "1.194.O._10N.286.54.B1\n",
        "1.269.O._10N.286.54.A6\n",
        "1.291.O._10N.286.55.F6\n",
        "1.105.O._10N.286.49.B4\n",
        "1.094.O._10N.286.55.E12\n",
        "1.141.A._10N.261.49.B3\n",
        "1.038.O._10N.286.51.C2\n",
        "1.154.O._10N.222.52.B12\n",
        "1.189.B._10N.286.51.B5\n",
        "1.139.B._10N.261.48.C6\n",
        "2.096.O._10N.286.48.B5\n",
        "1.027.O._10N.286.54.B8\n",
        "1.173.O._10N.261.55.A11\n",
        "1.176.O._10N.261.55.F5\n",
        "1.009.O._10N.261.51.C9\n",
        "1.031.O._10N.261.46.F8\n",
        "2.117.O._10N.261.45.E8\n",
        "1.253.O._10N.286.45.B12\n",
        "1.199.A._10N.286.55.C10\n",
        "1.072.O._10N.286.48.A12\n",
        "1.095.O._10N.286.46.E10\n",
        "1.177.O._10N.286.45.E10\n",
        "1.046.O._10N.286.52.E3\n",
        "1.225.O._10N.261.48.B7\n",
        "1.213.O._10N.222.54.F10\n",
        "1.107.B._10N.286.52.E10\n",
        "1.069.O._10N.286.49.F11\n",
        "1.028.O._10N.286.45.B6\n",
        "1.214.O._10N.222.54.F11\n",
        "1.036.O._10N.286.45.C3\n",
        "1.250.O._10N.261.55.E11\n",
        "1.276.O._10N.286.54.E4\n",
        "1.167.O._10N.261.51.F2\n",
        "1.211.A._10N.222.52.F11\n",
        "1.235.O._10N.261.52.B2\n",
        "1.081.O._10N.286.52.C2\n",
        "1.189.O._10N.286.51.B5\n",
        "2.275.O._10N.286.54.E11\n",
        "1.016.O._10N.286.46.A11\n",
        "1.093.O._10N.286.55.E10\n",
        "1.238.B._10N.261.52.F10\n",
        "1.121.O._10N.286.46.C4\n",
        "1.174.O._10N.261.55.A8\n",
        "1.243.O._10N.261.54.B5\n",
        "1.270.B._10N.286.54.A8\n",
        "2.044.O._10N.261.51.B8\n",
        "1.224.A._10N.261.48.B1\n",
        "1.148.O._10N.286.54.A10\n",
        "1.089.O._10N.261.51.F9\n",
        "1.198.B._10N.286.54.F4\n",
        "1.157.O._10N.261.45.B7\n",
        "1.088.O._10N.261.46.A1\n",
        "1.274.O._10N.286.54.E1\n",
        "1.070.O._10N.261.45.B2\n",
        "1.026.O._10N.222.49.C7\n",
        "1.127.O._10N.286.52.E12\n",
        "1.172.O._10N.261.52.F5\n",
        "1.271.A._10N.286.54.B4\n",
        "1.008.O._10N.286.54.E5\n",
        "1.102.O._10N.261.45.E3\n",
        "1.086.O._10N.222.51.F8\n",
        "1.085.O._10N.222.51.E3\n",
        "1.204.O._10N.222.46.F12\n",
        "1.023.O._10N.222.51.B4\n",
        "1.097.O._10N.286.49.B3\n",
        "1.050.O._10N.286.48.A6\n",
        "1.107.C._10N.286.52.E10\n",
        "1.013.O._10N.286.54.F9\n",
        "2.159.B._10N.261.46.F12\n",
        "1.025.O._10N.222.46.B6\n",
        "1.049.O._10N.286.54.B5\n",
        "1.080.O._10N.286.48.A4\n",
        "1.205.O._10N.222.51.A7\n",
        "1.077.O._10N.261.45.A10\n",
        "1.062.O._10N.286.55.C3\n",
        "1.074.O._10N.222.49.B7\n",
        "1.202.O._10N.222.45.E8\n",
        "1.263.B._10N.286.51.B1\n",
        "1.277.B._10N.286.54.E7\n",
        "1.123.O._10N.286.48.F3\n",
        "1.233.B._10N.261.51.E6\n",
        "1.216.O._10N.222.55.C12\n",
        "1.039.O._10N.286.55.A2\n",
        "1.175.O._10N.261.55.B3\n",
        "1.208.B._10N.222.52.A7\n",
        "1.110.O._10N.261.52.C1\n",
        "1.156.O._10N.261.45.A6\n",
        "1.064.O._10N.261.52.E5\n",
        "1.232.O._10N.261.51.E11\n",
        "1.178.O._10N.286.45.E12\n",
        "1.054.O._10N.261.52.A1\n",
        "1.215.A._10N.222.54.F7\n",
        "1.103.O._10N.261.52.F2\n",
        "1.236.O._10N.261.52.C4\n",
        "1.187.O._10N.286.49.F1\n",
        "1.119.O._10N.261.51.A9\n",
        "1.084.O._10N.261.49.F5\n",
        "1.043.O._10N.261.52.C7\n",
        "1.022.O._10N.286.45.A10\n",
        "1.151.O._10N.222.46.B1\n",
        "1.067.O._10N.261.52.C9\n",
        "1.189.C._10N.286.51.B5\n",
        "1.037.O._10N.261.52.F7\n",
        "1.251.O._10N.261.55.E5\n",
        "1.249.B._10N.261.55.B9\n",
        "1.268.A._10N.286.54.A11\n",
        "1.068.O._10N.261.51.F8\n",
        "1.112.O._10N.286.46.B11\n",
        "1.242.O._10N.261.54.B2\n",
        "1.201.B._10N.286.55.F1\n",
        "1.272.O._10N.286.54.C4\n",
        "1.181.O._10N.286.46.C9\n",
        "1.264.O._10N.286.51.F2\n",
        "1.198.A._10N.286.54.F4\n",
        "1.197.A._10N.286.54.F2\n",
        "1.075.O._10N.286.55.B10\n",
        "1.171.O._10N.261.52.F12\n",
        "1.122.A._10N.286.46.F8\n",
        "1.024.O._10N.261.45.F8\n",
        "1.149.O._10N.286.55.A12\n",
        "1.160.O._10N.261.48.B11\n",
        "1.257.O._10N.286.46.A4\n",
        "1.056.O._10N.261.48.C11\n",
        "3.058.O._10N.286.46.B8\n",
        "1.124.O._10N.286.49.B1\n",
        "1.249.A._10N.261.55.B9\n",
        "1.007.O._10N.261.55.F9\n",
        "1.055.O._10N.286.55.E9\n",
        "1.247.A._10N.261.54.E12\n",
        "1.118.A._10N.261.49.F6\n",
        "1.179.O._10N.286.45.F12\n",
        "1.117.O._10N.261.45.E8\n",
        "1.186.O._10N.286.49.E3\n",
        "1.020.O._10N.222.48.A2\n",
        "1.210.O._10N.222.52.C2\n",
        "1.079.O._10N.286.45.E9\n",
        "1.280.O._10N.286.54.F6\n",
        "1.283.A._10N.286.55.A1\n",
        "1.211.B._10N.222.52.F11\n",
        "1.083.O._10N.286.52.B9\n",
        "1.283.C._10N.286.55.A1\n",
        "1.048.O._10N.286.46.A10\n",
        "1.091.O._10N.286.52.B12\n",
        "1.228.O._10N.261.49.C1\n",
        "1.143.O._10N.261.55.C8\n",
        "1.122.B._10N.286.46.F8\n",
        "1.188.A._10N.286.51.A6\n",
        "1.237.A._10N.261.52.C5\n",
        "1.226.O._10N.261.48.E5\n",
        "1.270.A._10N.286.54.A8\n",
        "1.244.A._10N.261.54.C3\n",
        "12G01.O._12G01\n",
        "1.115.A._10N.222.49.B11\n",
        "1.188.B._10N.286.51.A6\n",
        "1.021.B._10N.222.51.F9\n",
        "1.057.O._10N.261.46.B12\n",
        "1.137.O._10N.261.46.B5\n",
        "1.190.O._10N.286.51.F12\n",
        "2.130.O._10N.222.46.C2\n",
        "1.125.O._10N.286.49.F5\n",
        "1.193.O._10N.286.52.C6\n",
        "1.159.O._10N.261.46.F12\n",
        "1.034.O._10N.261.46.B7\n",
        "1.106.O._10N.286.51.F7\n",
        "1.090.B._10N.286.48.F1\n",
        "1.183.O._10N.286.48.B7\n",
        "1.139.A._10N.261.48.C6\n",
        "1.273.O._10N.286.54.C7\n",
        "1.150.O._10N.222.46.A6\n",
        "1.066.O._10N.286.46.E8\n",
        "1.098.O._10N.286.51.B9\n",
        "1.071.A._10N.286.46.A12\n",
        "2.058.O._10N.286.46.B8\n",
        "1.032.O._10N.261.54.F5\n",
        "1.060.A._10N.261.48.B5\n",
        "1.042.O._10N.286.45.B8\n",
        "1.209.O._10N.222.52.B2\n",
        "1.138.O._10N.261.48.A1\n",
        "1.082.O._10N.261.49.E4\n",
        "1.268.B._10N.286.54.A11\n",
        "1.283.B._10N.286.55.A1\n",
        "1.233.A._10N.261.51.E6\n",
        "1.245.O._10N.261.54.C7\n",
        "2.159.A._10N.261.46.F12\n",
        "1.162.O._10N.261.48.E3\n",
        "1.266.O._10N.286.52.F9\n",
        "1.263.A._10N.286.51.B1\n",
        "1.188.C._10N.286.51.A6\n",
        "1.113.A._10N.286.51.E7\n",
        "1.135.O._10N.222.54.B6\n",
        "2.095.A._10N.286.46.E10\n",
        "1.021.C._10N.222.51.F9\n",
        "1.044.O._10N.261.51.B8\n",
        "1.168.O._10N.261.52.A10\n",
        "1.207.B._10N.222.51.C2\n",
        "1.061.O._10N.286.55.C2\n",
        "1.136.O._10N.261.45.E11\n",
        "1.184.A._10N.286.49.A5\n",
        "1.248.O._10N.261.54.F1\n",
        "1.144.O._10N.286.45.B3\n",
        "1.169.O._10N.261.52.B1\n",
        "1.107.A._10N.286.52.E10\n",
        "1.158.O._10N.261.45.E12\n",
        "1.246.O._10N.261.54.E10\n",
        "1.111.A._10N.286.45.E6\n",
        "1.034.X._10N.261.46.B7\n",
        "1.285.O._10N.286.55.C12\n",
        "1.116.O._10N.222.52.C10\n",
        "1.282.A._10N.286.54.F8\n",
        "1.256.O._10N.286.45.F8\n",
        "1.255.O._10N.286.45.F1\n",
        "1.142.O._10N.261.49.E11\n",
        "1.005.O._10N.286.48.F2\n",
        "1.231.O._10N.261.49.F8\n",
        "1.132.O._10N.222.49.F8\n",
        "1.131.O._10N.222.49.A8\n",
        "1.240.O._10N.261.52.F8\n",
        "1.111.B._10N.286.45.E6\n",
        "1.254.O._10N.286.45.C8\n",
        "1.040.O._10N.286.45.B9\n",
        "1.052.A._10N.286.46.C3\n",
        "1.033.O._10N.222.49.B8\n",
        "1.182.O._10N.286.46.E1\n"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "in terminal:\n",
      "bash annote_step1.sh"
     ]
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "rast-process-genome-batch --workflow vir_genomes.workflow2.json --verbose ToSubmit\n",
      "\n",
      "job ID: 8d2cee15-b72c-45a7-87c6-05c6e5ab9e3e\n",
      "\n",
      "rast-query-genome-batch 8d2cee15-b72c-45a7-87c6-05c6e5ab9e3e\n",
      "\n",
      "mkdir rasttk_annotation\n",
      "\n",
      "rast-download-genome-batch 8d2cee15-b72c-45a7-87c6-05c6e5ab9e3e rasttk_annotation"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "rast-process-genome <./ToSubmit/1.226.O.gto> ./rasttk_annotation/1.226.O.gto.gto\n",
      "rast-process-genome <./ToSubmit/1.266.O.gto> ./rasttk_annotation/1.266.O.gto.gto\n",
      "\n",
      "rast-export-genome genbank <1.226.O.gto.gto> 1.226.O.gbk\n",
      "rast-export-genome genbank <1.266.O.gto.gto> 1.266.O.gbk"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#rast-export-genome genbank <1.015.O.gto.gto> 1.015.O.gbk\n",
      "\n",
      "out=open(\"gto2gbk.commands.sh\",\"w\")\n",
      "out.write(\"mkdir gbk\\n\\n\")\n",
      "for line in finals:\n",
      "    phagename=line\n",
      "    out.write(\"rast-export-genome genbank <\"+phagename+\"gto.gto> \"+\"./gbk/\"+phagename+\"gbk\\n\\nsleep 3\\n\\n\")\n",
      "out.close()\n",
      "!open gto2gbk.commands.sh -a textedit"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      ">not all genomes were successfully converted"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "out=open(\"gto2gbk.commands.sh\",\"w\")\n",
      "\n",
      "for line in finals:\n",
      "    gbk=\"./rasttk_annotation/\"+line+\"gbk\"\n",
      "    if os.path.exists(gbk):\n",
      "        print(line+\" exists\")\n",
      "        if os.path.getsize(gbk)==0:\n",
      "            out.write(\"rast-export-genome --verbose genbank <\"+line+\"gto.gto> \"+line+\"gbk\\n\\nsleep 5\\n\\n\")\n",
      "    else:\n",
      "        out.write(\"rast-export-genome --verbose genbank <\"+line+\"gto.gto> \"+line+\"gbk\\n\\nsleep 5\\n\\n\")\n",
      "out.close()\n",
      "!open gto2gbk.commands.sh -a textedit"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1.206.O. exists\n",
        "1.237.B. exists\n",
        "1.267.O. exists\n",
        "1.015.O. exists\n",
        "1.100.O. exists\n",
        "1.239.O. exists\n",
        "1.196.O. exists\n",
        "1.134.O. exists\n",
        "1.278.O. exists\n",
        "1.215.B. exists\n",
        "1.289.A. exists\n",
        "1.293.O. exists\n",
        "1.199.B. exists\n",
        "1.087.A. exists\n",
        "1.217.O. exists\n",
        "1.223.O. exists\n",
        "1.108.O. exists\n",
        "1.259.O. exists\n",
        "1.115.B. exists\n",
        "2.095.B. exists\n",
        "1.262.O. exists\n",
        "1.155.O. exists\n",
        "1.195.O. exists\n",
        "1.277.A. exists\n",
        "1.017.O. exists\n",
        "1.104.O. exists\n",
        "1.166.O. exists\n",
        "1.170.O. exists\n",
        "1.238.A. exists\n",
        "1.275.O. exists\n",
        "1.133.O. exists\n",
        "1.185.O. exists\n",
        "2.092.O. exists\n",
        "1.030.O. exists\n",
        "1.247.B. exists\n",
        "1.003.O. exists\n",
        "1.165.O. exists\n",
        "1.271.B. exists\n",
        "1.219.O. exists\n",
        "1.191.O. exists\n",
        "1.011.O. exists\n",
        "1.118.B. exists\n",
        "1.101.O. exists\n",
        "1.047.O. exists\n",
        "1.161.O. exists\n",
        "1.286.O. exists\n",
        "1.164.O. exists\n",
        "1.021.A. exists\n",
        "1.029.O. exists\n",
        "1.076.O. exists\n",
        "1.200.O. exists\n",
        "1.152.O. exists\n",
        "1.281.O. exists\n",
        "1.012.O. exists\n",
        "1.147.O. exists\n",
        "1.261.O. exists\n",
        "1.284.A. exists\n",
        "1.287.O. exists\n",
        "1.063.O. exists\n",
        "1.265.O. exists\n",
        "1.004.O. exists\n",
        "1.194.O. exists\n",
        "1.269.O. exists\n",
        "1.291.O. exists\n",
        "1.105.O. exists\n",
        "1.094.O. exists\n",
        "1.141.A. exists\n",
        "1.038.O. exists\n",
        "1.154.O. exists\n",
        "1.189.B. exists\n",
        "1.139.B. exists\n",
        "2.096.O. exists\n",
        "1.027.O. exists\n",
        "1.173.O. exists\n",
        "1.176.O. exists\n",
        "1.009.O. exists\n",
        "1.031.O. exists\n",
        "2.117.O. exists\n",
        "1.253.O. exists\n",
        "1.199.A. exists\n",
        "1.072.O. exists\n",
        "1.095.O. exists\n",
        "1.177.O. exists\n",
        "1.046.O. exists\n",
        "1.225.O. exists\n",
        "1.213.O. exists\n",
        "1.107.B. exists\n",
        "1.069.O. exists\n",
        "1.028.O. exists\n",
        "1.214.O. exists\n",
        "1.036.O. exists\n",
        "1.250.O. exists\n",
        "1.276.O. exists\n",
        "1.167.O. exists\n",
        "1.211.A. exists\n",
        "1.235.O. exists\n",
        "1.081.O. exists\n",
        "1.189.O. exists\n",
        "2.275.O. exists\n",
        "1.016.O. exists\n",
        "1.093.O. exists\n",
        "1.238.B. exists\n",
        "1.121.O. exists\n",
        "1.174.O. exists\n",
        "1.243.O. exists\n",
        "1.270.B. exists\n",
        "2.044.O. exists\n",
        "1.224.A. exists\n",
        "1.148.O. exists\n",
        "1.089.O. exists\n",
        "1.198.B. exists\n",
        "1.157.O. exists\n",
        "1.088.O. exists\n",
        "1.274.O. exists\n",
        "1.070.O. exists\n",
        "1.026.O. exists\n",
        "1.127.O. exists\n",
        "1.172.O. exists\n",
        "1.271.A. exists\n",
        "1.008.O. exists\n",
        "1.102.O. exists\n",
        "1.086.O. exists\n",
        "1.085.O. exists\n",
        "1.204.O. exists\n",
        "1.023.O. exists\n",
        "1.097.O. exists\n",
        "1.050.O. exists\n",
        "1.107.C. exists\n",
        "1.013.O. exists\n",
        "2.159.B. exists\n",
        "1.025.O. exists\n",
        "1.049.O. exists\n",
        "1.080.O. exists\n",
        "1.205.O. exists\n",
        "1.077.O. exists\n",
        "1.062.O. exists\n",
        "1.074.O. exists\n",
        "1.202.O. exists\n",
        "1.263.B. exists\n",
        "1.277.B. exists\n",
        "1.123.O. exists\n",
        "1.233.B. exists\n",
        "1.216.O. exists\n",
        "1.039.O. exists\n",
        "1.175.O. exists\n",
        "1.208.B. exists\n",
        "1.110.O. exists\n",
        "1.156.O. exists\n",
        "1.064.O. exists\n",
        "1.232.O. exists\n",
        "1.178.O. exists\n",
        "1.054.O. exists\n",
        "1.215.A. exists\n",
        "1.103.O. exists\n",
        "1.236.O. exists\n",
        "1.187.O. exists\n",
        "1.119.O. exists\n",
        "1.084.O. exists\n",
        "1.043.O. exists\n",
        "1.022.O. exists\n",
        "1.151.O. exists\n",
        "1.067.O. exists\n",
        "1.189.C. exists\n",
        "1.037.O. exists\n",
        "1.251.O. exists\n",
        "1.249.B. exists\n",
        "1.268.A. exists\n",
        "1.068.O. exists\n",
        "1.112.O. exists\n",
        "1.242.O. exists\n",
        "1.201.B. exists\n",
        "1.272.O. exists\n",
        "1.181.O. exists\n",
        "1.264.O. exists\n",
        "1.198.A. exists\n",
        "1.197.A. exists\n",
        "1.075.O. exists\n",
        "1.171.O. exists\n",
        "1.122.A. exists\n",
        "1.024.O. exists\n",
        "1.149.O. exists\n",
        "1.160.O. exists\n",
        "1.257.O. exists\n",
        "1.056.O. exists\n",
        "3.058.O. exists\n",
        "1.124.O. exists\n",
        "1.249.A. exists\n",
        "1.007.O. exists\n",
        "1.055.O. exists\n",
        "1.247.A. exists\n",
        "1.118.A. exists\n",
        "1.179.O. exists\n",
        "1.117.O. exists\n",
        "1.186.O. exists\n",
        "1.020.O. exists\n",
        "1.210.O. exists\n",
        "1.079.O. exists\n",
        "1.280.O. exists\n",
        "1.283.A. exists\n",
        "1.211.B. exists\n",
        "1.083.O. exists\n",
        "1.283.C. exists\n",
        "1.048.O. exists\n",
        "1.091.O. exists\n",
        "1.228.O. exists\n",
        "1.143.O. exists\n",
        "1.122.B. exists\n",
        "1.188.A. exists\n",
        "1.237.A. exists\n",
        "1.226.O. exists\n",
        "1.270.A. exists\n",
        "1.244.A. exists\n",
        "12G01.O. exists\n",
        "1.115.A. exists\n",
        "1.188.B. exists\n",
        "1.021.B. exists\n",
        "1.057.O. exists\n",
        "1.137.O. exists\n",
        "1.190.O. exists\n",
        "2.130.O. exists\n",
        "1.125.O. exists\n",
        "1.193.O. exists\n",
        "1.159.O. exists\n",
        "1.034.O. exists\n",
        "1.106.O. exists\n",
        "1.090.B. exists\n",
        "1.183.O. exists\n",
        "1.139.A. exists\n",
        "1.273.O. exists\n",
        "1.150.O. exists\n",
        "1.066.O. exists\n",
        "1.098.O. exists\n",
        "1.071.A. exists\n",
        "2.058.O. exists\n",
        "1.032.O. exists\n",
        "1.060.A. exists\n",
        "1.042.O. exists\n",
        "1.209.O. exists\n",
        "1.138.O. exists\n",
        "1.082.O. exists\n",
        "1.268.B. exists\n",
        "1.283.B. exists\n",
        "1.233.A. exists\n",
        "1.245.O. exists\n",
        "2.159.A. exists\n",
        "1.162.O. exists\n",
        "1.266.O. exists\n",
        "1.263.A. exists\n",
        "1.188.C. exists\n",
        "1.113.A. exists\n",
        "1.135.O. exists\n",
        "2.095.A. exists\n",
        "1.021.C. exists\n",
        "1.044.O. exists\n",
        "1.168.O. exists\n",
        "1.207.B. exists\n",
        "1.061.O. exists\n",
        "1.136.O. exists\n",
        "1.184.A. exists\n",
        "1.248.O. exists\n",
        "1.144.O. exists\n",
        "1.169.O. exists\n",
        "1.107.A. exists\n",
        "1.158.O. exists\n",
        "1.246.O. exists\n",
        "1.111.A. exists\n",
        "1.034.X. exists\n",
        "1.285.O. exists\n",
        "1.116.O. exists\n",
        "1.282.A. exists\n",
        "1.256.O. exists\n",
        "1.255.O. exists\n",
        "1.142.O. exists\n",
        "1.005.O. exists\n",
        "1.231.O. exists\n",
        "1.132.O. exists\n",
        "1.131.O. exists\n",
        "1.240.O. exists\n",
        "1.111.B. exists\n",
        "1.254.O. exists\n",
        "1.040.O. exists\n",
        "1.052.A. exists\n",
        "1.033.O. exists\n",
        "1.182.O. exists\n"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "in terminal:\n",
      "bash gto2gbk.commands.sh"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#to extract cds from rasttk gbk files into fna and faa files:\n",
      "from Bio import SeqIO\n",
      "import re\n",
      "from Bio.SeqFeature import FeatureLocation\n",
      "\n",
      "def cds_rasttk_gbk_extract(list_of_files):\n",
      "    gbkfiles=open(list_of_files)\n",
      "    genbankfiles=gbkfiles.readlines()\n",
      "    for line in genbankfiles:\n",
      "        filename=line.replace(\".gbk\\n\",\"\")\n",
      "        cdsfna=open(filename+\".fna\", \"w\")\n",
      "        cdsfaa=open(filename+\".faa\", \"w\")\n",
      "        gbk=line.replace(\"\\n\",\"\")\n",
      "        for rec in SeqIO.parse(gbk, \"genbank\"):\n",
      "            for feature in rec.features:\n",
      "                if feature.type==\"CDS\":\n",
      "                    name=str(feature.qualifiers[\"db_xref\"])+\"_\"+str(feature.qualifiers[\"product\"])\n",
      "                    local=feature.location                 \n",
      "                    if bool(re.search(r\"\\-\",str(local))):    #if the feature location is on the negative strand\n",
      "                        stop=local.start                     #the start location should actually be the end location\n",
      "                        start=local.end                      #and the end location should be the start location \n",
      "                        feature.location=FeatureLocation(start+1, stop+1, strand=-1)   #re-writing the feature location in the correct format\n",
      "                        na=feature.location.extract(rec).seq                   #extract the nucleic acid sequence of this feature\n",
      "                    else:\n",
      "                        na=feature.location.extract(rec).seq    #if it's on the positive strand the feature location is written in the correct way\n",
      "                    aa=str(feature.qualifiers[\"translation\"])\n",
      "                    aa=aa.replace(\"[\",\"\").replace(\"]\",\"\").replace(\"'\",\"\")\n",
      "                    cdsfna.write(\">\"+filename+\"_\"+str(name)+\"\\n\"+str(na)+\"\\n\")\n",
      "                    cdsfaa.write(\">\"+filename+\"_\"+str(name)+\"\\n\"+str(aa)+\"\\n\")\n",
      "        cdsfna.close()\n",
      "        cdsfaa.close()    \n",
      "          #  feature_name=feature.qualifiers[\"db_xref\"]\n",
      "           # print(feature_name+str(count))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 17
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#to extract cds from gbk files into fna and faa files:\n",
      "from Bio import SeqIO\n",
      "import re\n",
      "from Bio.SeqFeature import FeatureLocation\n",
      "\n",
      "def cds_gbk_extract(list_of_files):\n",
      "    gbkfiles=open(list_of_files)\n",
      "    genbankfiles=gbkfiles.readlines()\n",
      "    for line in genbankfiles:\n",
      "        filename=line.replace(\".gbk\\n\",\"\")\n",
      "        cdsfna=open(filename+\".fna\", \"w\")\n",
      "        cdsfaa=open(filename+\".faa\", \"w\")\n",
      "        gbk=line.replace(\"\\n\",\"\")\n",
      "        for rec in SeqIO.parse(gbk, \"genbank\"):\n",
      "            genome=SeqIO.convert(line.rstrip(\"\\n\"), \"genbank\", filename+\".genome.fasta\", \"fasta\")\n",
      "            for feature in rec.features:\n",
      "                if feature.type==\"CDS\":\n",
      "                    name=str(feature.qualifiers[\"locus_tag\"])#+\"_\"+str(feature.qualifiers[\"product\"])\n",
      "                    local=feature.location                 \n",
      "                    na=feature.location.extract(rec).seq   \n",
      "                    aa=str(feature.qualifiers[\"translation\"])\n",
      "                    aa=aa.replace(\"[\",\"\").replace(\"]\",\"\").replace(\"'\",\"\")\n",
      "                    cdsfna.write(\">\"+str(name)+\"\\n\"+str(na)+\"\\n\")\n",
      "                    cdsfaa.write(\">\"+str(name)+\"\\n\"+str(aa)+\"\\n\")\n",
      "        cdsfna.close()\n",
      "        cdsfaa.close()    \n",
      "          #  feature_name=feature.qualifiers[\"db_xref\"]\n",
      "           # print(feature_name+str(count))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 29
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pwd"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 45,
       "text": [
        "u'/Users/jmb/Desktop/ViralFate/PhageGenomeAssembly'"
       ]
      }
     ],
     "prompt_number": 45
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import os\n",
      "os.chdir(\"./Ref_Phage_Genomes/\")\n",
      "!ls *gbk > gbk.files.txt\n",
      "cds_gbk_extract(\"gbk.files.txt\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 46
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "ls"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "ColiphageN4.faa       VibriophageJA1.fna    VibriophageVBP47.gbk\r\n",
        "ColiphageN4.fna       VibriophageJA1.gbk    gbk.files.txt\r\n",
        "ColiphageN4.gbk       VibriophageVBP47.faa\r\n",
        "VibriophageJA1.faa    VibriophageVBP47.fna\r\n"
       ]
      }
     ],
     "prompt_number": 25
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import os\n",
      "os.chdir(\"../genome_seqs/\")\n",
      "!ls *.gbk > gbk.files.txt\n",
      "cds_rasttk_gbk_extract(\"gbk.files.txt\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "scp ./N4Like_Refs/*genome.fasta jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/group5/finished/N4Like_Refs/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!scp  ./blast1_for_ACT.sh jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/group5/finished/blast_comparisons/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "blast1_for_ACT.sh                               0%    0     0.0KB/s   --:-- ETA\r",
        "blast1_for_ACT.sh                             100% 1235     1.2KB/s   00:00    \r\n"
       ]
      }
     ],
     "prompt_number": 42
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import os\n",
      "os.chdir(\"./PhageGenome/group6\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "OSError",
       "evalue": "[Errno 2] No such file or directory: './group6'",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-50-06546f8fc248>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"./group6\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mOSError\u001b[0m: [Errno 2] No such file or directory: './group6'"
       ]
      }
     ],
     "prompt_number": 50
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pwd"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 34,
       "text": [
        "u'/Users/jmb/Desktop/ViralFate'"
       ]
      }
     ],
     "prompt_number": 34
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "#!/bin/bash                                                                                                                                                                \n",
      "#SBATCH -n 16                                                                                                                                                              \n",
      "#SBATCH -N 1                                                                                                                     \n",
      "#SBATCH -o db.out      # File to which STDOUT will be written                                                                                   \n",
      "#SBATCH -e db.err      # File to which STDERR will be written                                                                                   \n",
      "#SBATCH -p newnodes                # name of partition to use  \n",
      "#saved as blast1_for_ACT.sh\n",
      "\n",
      "module add engaging/ncbi-blast/2.2.30+\n",
      "\n",
      "makeblastdb -in ColiphageN4.genome.fasta -dbtype nucl \n",
      "makeblastdb -in VibriophageVBP47.genome.fasta -dbtype nucl\n",
      "makeblastdb -in VibriophageJA1.genome.fasta -dbtype nucl\n",
      "\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.026.O.step2.fasta -outfmt 6  -out 1.026.O_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.025.O.step1.fasta -outfmt 6  -out 1.025.O_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.097.O.step2.fasta -outfmt 6  -out 1.097.O_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.188.C.step2.fasta -outfmt 6  -out 1.188.C_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.188.B.step2.fasta -outfmt 6  -out 1.188.B_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.169.O.step1.fasta -outfmt 6  -out 1.169.O_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../1.261.O.step2.fasta -outfmt 6  -out 1.261.O_vs_coliphageN4.txt\n",
      "tblastx -db ColiphageN4.genome.fasta -query ../2.130.O.step2.fasta -outfmt 6  -out 2.130.O_vs_coliphageN4.txt"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 0
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}