{
 "metadata": {
  "name": "",
  "signature": "sha256:31852e681b854c1ae5fff8c1ffa30e5a9125182908ca6971d2960ffb3f244f3c"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "###Want to check out BBMap's \"tadpole\" assembler to see how it performs in comparison to some curated assemblies.\n",
      ">Focus on: group 4 and group 5\n",
      "    \n",
      "    >>Group 4 appears to have variable genome lengths (repeats?)\n",
      "    \n",
      "    >>Group 5 has a very specific and obvious assembly problem (terminal repeats)\n",
      "    \n",
      "    >>Group 15 because of variable repeats at the ends"
     ]
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "#BBMap's tadpole.sh documentation:\n",
      "\n",
      "bash tadpole.sh\n",
      "\n",
      "Written by Brian Bushnell\n",
      "Last modified August 8, 2015\n",
      "\n",
      "Description:  Uses kmer counts to assembles contigs, extend sequences, \n",
      "or error-correct reads.  Tadpole has no upper bound for kmer length,\n",
      "but some values are not supported.  Specifically, it allows 1-31,\n",
      "multiples of 2 from 32-62, multiples of 3 from 63-93, etc.\n",
      "\n",
      "Usage:\n",
      "Generation:   tadpole.sh in=<reads> out=<contigs>\n",
      "Extension:    tadpole.sh in=<reads> ine=<sequences> oute=<extended> mode=extend\n",
      "Correction:   tadpole.sh in=<reads> out=<corrected> mode=correct\n",
      "\n",
      "Input may be fasta or fastq, compressed or uncompressed.\n",
      "\n",
      "\n",
      "Input parameters:\n",
      "in=<file>           Primary input file for reads to use as kmer data.\n",
      "in2=<file>          Second input file for paired data.\n",
      "extend=<file>       Primary input file for sequences to extend.\n",
      "extend2=<file>      Second input file for paired reads.\n",
      "reads=-1            Only process this number of reads, then quit (-1 means all).\n",
      "\n",
      "Output parameters:\n",
      "out=<file>          Write contigs (in contig mode).\n",
      "oute=<file>         Write extended reads (in extend mode).\n",
      "ihist=<file>        Write insert size histogram (in insert mode).\n",
      "dump=<file>         Write kmers and their counts.\n",
      "fastadump=t         Write kmers and counts as fasta versus 2-column tsv.\n",
      "mincounttodump=1    Only dump kmers with at least this depth.\n",
      "showstats=t         Print assembly statistics after writing contigs.\n",
      "\n",
      "Hashing parameters:\n",
      "k=31                Kmer length (1 to infinity).  Time and memory use increase with larger K, in increments of 31.\n",
      "prealloc=t          Pre-allocate memory rather than dynamically growing; faster and more memory-efficient.  A float fraction (0-1) may be specified, default 1.\n",
      "prefilter=0         If set to a positive integer, use a countmin sketch to ignore kmers with depth of that value or lower.\n",
      "prehashes=2         Number of hashes for prefilter.\n",
      "prefiltersize=0.2   Fraction of memory to use for prefilter.\n",
      "minq=6              Ignore kmers containing bases with quality below this. (TODO)\n",
      "minprob=0.5         Ignore kmers with overall probability of correctness below this.\n",
      "threads=X           Spawn X hashing threads (default is number of logical processors).\n",
      "onepass=f           If true, prefilter will be generated in same pass as kmer counts.  Much faster but counts will be lower, by up to prefilter's depth limit.\n",
      "rcomp=t             Store and count each kmer together and its reverse-complement.\n",
      "\n",
      "Assembly parameters:\n",
      "mindepthseed=3      (mds) Minimum kmer depth to seed a new contig or begin extension.\n",
      "mindepthextend=2    (mde) Minimum kmer depth continue extension of a read or contig.\n",
      "branchmult1=20      (bm1) Min ratio of 1st to 2nd-greatest path depth at high depth.\n",
      "branchmult2=3       (bm2) Min ratio of 1st to 2nd-greatest path depth at low depth.\n",
      "branchlower=3       (blc) Max value of 2nd-greatest path depth to be considered low.\n",
      "minextension=1      (mine) Do not keep contigs that did not extend at least this much.\n",
      "mincontig=200       (minc) Do not write contigs shorter than this.\n",
      "mincoverage=1       (mincov) Do not write contigs with average coverage below this.\n",
      "contigpasses=16     Build contigs with decreasing seed depth for this many iterations.\n",
      "contigpassmult=1.7  Ratio between seed depth of two iterations.\n",
      "ownership=auto      For concurrency; do not touch.\n",
      "\n",
      "Processing modes:\n",
      "mode=contig         contig: Make contigs from kmers.\n",
      "                    extend: Extend sequences to be longer, and optionally\n",
      "                            perform error correction.\n",
      "                    correct: Error correct only.\n",
      "                    insert: Measure insert sizes.\n",
      "\n",
      "Extension parameters:\n",
      "extendleft=100      (el) Extend to the left by at most this many bases.\n",
      "extendright=100     (er) Extend to the right by at most this many bases.\n",
      "ibb=t               (ignorebackbranches) Do not stop at backward branches.\n",
      "\n",
      "Error-correction parameters:\n",
      "ecc=f               Error correct via kmer counts.\n",
      "pincer=t            If ecc is enabled, use the pincer algorithm.\n",
      "tail=t              If ecc is enabled, use the tail algorithm.\n",
      "aggressive=f        (aecc) Correct with bidirectional double-pass.\n",
      "markbadbases=0      (mbb) Any base fully covered by kmers with count\n",
      "                    below this will be changed to N.\n",
      "markdeltaonly=t     (mdo) Only mark bad bases adjacent to good bases.\n",
      "errormult1=60       (em1) Min ratio between kmer depths to call an error.\n",
      "errormult2=3        (em2) Alternate ratio between kmer depths.\n",
      "errorlowerconst=3   (elc) Use mult2 when the lower kmer is at most this deep.\n",
      "mincountcorrect=4   (mcc) Don't correct to kmers with count under this.\n",
      "pathsimilarityfraction=0.3 (psf) Max difference ratio considered similar.\n",
      "                           Controls whether a path appears to be continuous.\n",
      "pathsimilarityconstant=3   (psc) Absolute differences below this are ignored.\n",
      "errorextensionpincer=5     (eep) Verify this many additional bases after the\n",
      "                           correction as matching current bases, for pincer.\n",
      "errorextensiontail=9       (eet) Verify this many additional bases after the\n",
      "                           correction as matching current bases, for tail.\n",
      "\n",
      "Shaving parameters:\n",
      "shave=f             Remove dead ends (aka hair).\n",
      "rinse=f             Remove bubbles.\n",
      "maxshavedepth=1     (msd) Shave or rinse kmers at most this deep.\n",
      "maxshavedepth=1     (msd) Shave or rinse kmers at most this deep.\n",
      "exploredist=100     (sed) Quit after exploring this far.\n",
      "discardlength=150   (sdl) Discard shavings up to this long.\n",
      "\n",
      "\n",
      "Overlap parameters (for overlapping paired-end reads only):\n",
      "merge=f             Attempt to merge reads before counting kmers.\n",
      "ecco=f              Error correct via overlap, but do not merge reads.\n",
      "\n",
      "Java Parameters:\n",
      "-Xmx                This will be passed to Java to set memory usage, overriding the program's automatic memory detection.\n",
      "                    -Xmx20g will specify 20 gigs of RAM, and -Xmx200m will specify 200 megs.  The max is typically 85% of physical memory."
     ]
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "bash bbmap.sh\n",
      "\n",
      "BBMap v35.x\n",
      "Written by Brian Bushnell, from Dec. 2010 - present\n",
      "Last modified August 8, 2015\n",
      "\n",
      "Description:  Fast and accurate splice-aware read aligner.\n",
      "\n",
      "To index:     bbmap.sh ref=<reference fasta>\n",
      "To map:       bbmap.sh in=<reads> out=<output sam>\n",
      "To map without writing an index:\n",
      "    bbmap.sh ref=<reference fasta> in=<reads> out=<output sam> nodisk\n",
      "\n",
      "in=stdin will accept reads from standard in, and out=stdout will write to \n",
      "standard out, but file extensions are still needed to specify the format of the \n",
      "input and output files e.g. in=stdin.fa.gz will read gzipped fasta from \n",
      "standard in; out=stdout.sam.gz will write gzipped sam.\n",
      "\n",
      "Indexing Parameters (required when building the index):\n",
      "\n",
      "nodisk=f                Set to true to build index in memory and write nothing \n",
      "                        to disk except output.\n",
      "ref=<file>              Specify the reference sequence.  Only do this ONCE, \n",
      "                        when building the index (unless using 'nodisk').\n",
      "build=1                 If multiple references are indexed in the same directory,\n",
      "                        each needs a unique numeric ID (unless using 'nodisk').\n",
      "k=13                    Kmer length, range 8-15.  Longer is faster but uses \n",
      "                        more memory.  Shorter is more sensitive.\n",
      "path=<.>                Specify the location to write the index, if you don't \n",
      "                        want it in the current working directory.\n",
      "usemodulo=f             Throw away ~80% of kmers based on remainder modulo a \n",
      "                        number (reduces RAM by 50% and sensitivity slightly).\n",
      "                        Should be enabled both when building the index AND \n",
      "                        when mapping.\n",
      "rebuild=f               Force a rebuild of the index (ref= should be set).\n",
      "\n",
      "Input Parameters:\n",
      "\n",
      "build=1                 Designate index to use.  Corresponds to the number \n",
      "                        specified when building the index.\n",
      "in=<file>               Primary reads input; required parameter.\n",
      "in2=<file>              For paired reads in two files.\n",
      "interleaved=auto        True forces paired/interleaved input; false forces \n",
      "                        single-ended mapping. If not specified, interleaved \n",
      "                        status will be autodetected from read names.\n",
      "fastareadlen=500        Break up FASTA reads longer than this.  Max is 500 for\n",
      "                        BBMap and 6000 for BBMapPacBio.  Only works for FASTA\n",
      "                        input (use 'maxlen' for FASTQ input).\n",
      "unpigz=f                Spawn a pigz (parallel gzip) process for faster \n",
      "                        decompression than using Java.  \n",
      "                        Requires pigz to be installed.\n",
      "touppercase=t           (tuc) Convert lowercase letters in reads to upper case \n",
      "                        (otherwise they will not match the reference).\n",
      "\n",
      "Sampling Parameters:\n",
      "\n",
      "reads=-1                Set to a positive number N to only process the first N\n",
      "                        reads (or pairs), then quit.  -1 means use all reads.\n",
      "samplerate=1            Set to a number from 0 to 1 to randomly select that\n",
      "                        fraction of reads for mapping. 1 uses all reads.\n",
      "skipreads=0             Set to a number N to skip the first N reads (or pairs), \n",
      "                        then map the rest.\n",
      "\n",
      "Mapping Parameters:\n",
      "\n",
      "fast=f                  This flag is a macro which sets other paramters to run \n",
      "                        faster, at reduced sensitivity.  Bad for RNA-seq.\n",
      "slow=f                  This flag is a macro which sets other paramters to run \n",
      "                        slower, at greater sensitivity.  'vslow' is even slower.\n",
      "maxindel=16000          Don't look for indels longer than this. Lower is faster.\n",
      "                        Set to >=100k for RNAseq with long introns like mammals.\n",
      "strictmaxindel=f        When enabled, do not allow indels longer than 'maxindel'.\n",
      "                        By default these are not sought, but may be found anyway.\n",
      "minid=0.76              Approximate minimum alignment identity to look for. \n",
      "                        Higher is faster and less sensitive.\n",
      "minhits=1               Minimum number of seed hits required for candidate sites.\n",
      "                        Higher is faster.\n",
      "k=13                    Kmer length of index.  Lower is slower, more sensitive,\n",
      "                        and uses slightly less RAM.  Range is 7 to 15.\n",
      "local=f                 Set to true to use local, rather than global, alignments.\n",
      "                        This will soft-clip ugly ends of poor alignments.\n",
      "perfectmode=f           Allow only perfect mappings when set to true (very fast).\n",
      "semiperfectmode=f       Allow only perfect and semiperfect (perfect except for \n",
      "                        N's in the reference) mappings.\n",
      "threads=auto            (t) Set to number of threads desired.  By default, uses \n",
      "                        all cores available.\n",
      "ambiguous=best          (ambig) Set behavior on ambiguously-mapped reads (with \n",
      "                        multiple top-scoring mapping locations).\n",
      "                            best    (use the first best site)\n",
      "                            toss    (consider unmapped)\n",
      "                            random  (select one top-scoring site randomly)\n",
      "                            all     (retain all top-scoring sites)\n",
      "samestrandpairs=f       (ssp) Specify whether paired reads should map to the\n",
      "                        same strand or opposite strands.\n",
      "requirecorrectstrand=t  (rcs) Forbid pairing of reads without correct strand \n",
      "                        orientation.  Set to false for long-mate-pair libraries.\n",
      "killbadpairs=f          (kbp) If a read pair is mapped with an inappropriate\n",
      "                        insert size or orientation, the read with the lower  \n",
      "                        mapping quality is marked unmapped.\n",
      "pairedonly=f            (po) Treat unpaired reads as unmapped.  Thus they will \n",
      "                        be sent to 'outu' but not 'outm'.\n",
      "rcomp=f                 Reverse complement both reads prior to mapping (for LMP\n",
      "                        outward-facing libraries).\n",
      "rcompmate=f             Reverse complement read2 prior to mapping.\n",
      "pairlen=32000           Set max allowed distance between paired reads.  \n",
      "                        (insert size)=(pairlen)+(read1 length)+(read2 length)\n",
      "rescuedist=1200         Don't try to rescue paired reads if avg. insert size\n",
      "                        greater than this.  Lower is faster.\n",
      "rescuemismatches=32     Maximum mismatches allowed in a rescued read.  Lower\n",
      "                        is faster.\n",
      "bandwidthratio=0        (bwr) If above zero, restrict alignment band to this \n",
      "                        fraction of read length.  Faster but less accurate.\n",
      "usejni=f                (jni) Do alignments faster, in C code.  Requires \n",
      "                        compiling the C code; details are in /jni/README.txt.\n",
      "maxsites2=800           Don't analyze (or print) more than this many alignments \n",
      "                        per read.\n",
      "monitor=f               Kill this process if CPU usage drops to zero for\n",
      "                        a long time.  monitor=600,0.01 would kill after 600\n",
      "                        seconds under 1% usage.\n",
      "\n",
      "Quality and Trimming Parameters:\n",
      "\n",
      "qin=auto                Set to 33 or 64 to specify input quality value ASCII\n",
      "                        offset. 33 is Sanger, 64 is old Solexa.\n",
      "qout=auto               Set to 33 or 64 to specify output quality value ASCII \n",
      "                        offset (only if output format is fastq).\n",
      "qtrim=f                 Quality-trim ends before mapping.  Options are: \n",
      "                        'f' (false), 'l' (left), 'r' (right), and 'lr' (both).\n",
      "untrim=f                Undo trimming after mapping.  Untrimmed bases will be \n",
      "                        soft-clipped in cigar strings.\n",
      "trimq=6                 Trim regions with average quality below this \n",
      "                        (phred algorithm).\n",
      "mintrimlength=60        (mintl) Don't trim reads to be shorter than this.\n",
      "fakefastaquality=-1     (ffq) Set to a positive number 1-50 to generate fake\n",
      "                        quality strings for fasta input reads.\n",
      "ignorebadquality=f      (ibq) Keep going, rather than crashing, if a read has \n",
      "                        out-of-range quality values.\n",
      "usequality=t            Use quality scores when determining which read kmers\n",
      "                        to use as seeds.\n",
      "minaveragequality=0     (maq) Discard reads with average quality below this.\n",
      "maqb=0                  If positive, calculate maq from this many initial bases.\n",
      "\n",
      "Output Parameters:\n",
      "\n",
      "out=<file>              Write all reads to this file.\n",
      "outu=<file>             Write only unmapped reads to this file.  Does not \n",
      "                        include unmapped paired reads with a mapped mate.\n",
      "outm=<file>             Write only mapped reads to this file.  Includes \n",
      "                        unmapped paired reads with a mapped mate.\n",
      "mappedonly=f            If true, treats 'out' like 'outm'.\n",
      "bamscript=<file>        (bs) Write a shell script to <file> that will turn \n",
      "                        the sam output into a sorted, indexed bam file.\n",
      "ordered=f               Set to true to output reads in same order as input.  \n",
      "                        Slower and uses more memory.\n",
      "overwrite=f             (ow) Allow process to overwrite existing files.\n",
      "secondary=f             Print secondary alignments.\n",
      "sssr=0.95               (secondarysitescoreratio) Print only secondary alignments\n",
      "                        with score of at least this fraction of primary.\n",
      "ssao=f                  (secondarysiteasambiguousonly) Only print secondary \n",
      "                        alignments for ambiguously-mapped reads.\n",
      "maxsites=5              Maximum number of total alignments to print per read.\n",
      "                        Only relevant when secondary=t.\n",
      "quickmatch=f            Generate cigar strings more quickly.\n",
      "trimreaddescriptions=f  (trd) Truncate read and ref names at the first whitespace,\n",
      "                        assuming that the remainder is a comment or description.\n",
      "ziplevel=2              (zl) Compression level for zip or gzip output.\n",
      "pigz=f                  Spawn a pigz (parallel gzip) process for faster \n",
      "                        compression than Java.  Requires pigz to be installed.\n",
      "machineout=f            Set to true to output statistics in machine-friendly \n",
      "                        'key=value' format.\n",
      "printunmappedcount=f    Print the total number of unmapped reads and bases.\n",
      "                        If input is paired, the number will be of pairs\n",
      "                        for which both reads are unmapped.\n",
      "showprogress=0          If positive, print a '.' every X reads.\n",
      "showprogress2=0         If positive, print the number of seconds since the\n",
      "                        last progress update (instead of a '.').\n",
      "\n",
      "Post-Filtering Parameters:\n",
      "\n",
      "idfilter=0              Independant of minid; sets exact minimum identity \n",
      "                        allowed for alignments to be printed.  Range 0 to 1.\n",
      "subfilter=-1            Ban alignments with more than this many substitutions.\n",
      "insfilter=-1            Ban alignments with more than this many insertions.\n",
      "delfilter=-1            Ban alignments with more than this many deletions.\n",
      "indelfilter=-1          Ban alignments with more than this many indels.\n",
      "editfilter=-1           Ban alignments with more than this many edits.\n",
      "inslenfilter=-1         Ban alignments with an insertion longer than this.\n",
      "dellenfilter=-1         Ban alignments with a deletion longer than this.\n",
      "\n",
      "Sam flags and settings:\n",
      "\n",
      "noheader=f              Disable generation of header lines.\n",
      "sam=1.4                 Set to 1.4 to write Sam version 1.4 cigar strings, \n",
      "                        with = and X, or 1.3 to use M.\n",
      "saa=t                   (secondaryalignmentasterisks) Use asterisks instead of\n",
      "                        bases for sam secondary alignments.\n",
      "cigar=t                 Set to 'f' to skip generation of cigar strings (faster).\n",
      "keepnames=f             Keep original names of paired reads, rather than \n",
      "                        ensuring both reads have the same name.\n",
      "intronlen=999999999     Set to a lower number like 10 to change 'D' to 'N' in \n",
      "                        cigar strings for deletions of at least that length.\n",
      "rgid=                   Set readgroup ID.  All other readgroup fields \n",
      "                        can be set similarly, with the flag rgXX=\n",
      "mdtag=f                 Write MD tags.\n",
      "nhtag=f                 Write NH tags.\n",
      "xmtag=f                 Write XM tags (may only work correctly with ambig=all).\n",
      "amtag=f                 Write AM tags.\n",
      "nmtag=f                 Write NM tags.\n",
      "xstag=f                 Set to 'xs=fs', 'xs=ss', or 'xs=us' to write XS tags \n",
      "                        for RNAseq using firststrand, secondstrand, or \n",
      "                        unstranded libraries.  Needed by Cufflinks.  \n",
      "                        JGI mainly uses 'firststrand'.\n",
      "stoptag=f               Write a tag indicating read stop location, prefixed by YS:i:\n",
      "lengthtag=f             Write a tag indicating (query,ref) alignment lengths, \n",
      "                        prefixed by YL:Z:\n",
      "idtag=f                 Write a tag indicating percent identity, prefixed by YI:f:\n",
      "inserttag=f             Write a tag indicating insert size, prefixed by X8:Z:\n",
      "scoretag=f              Write a tag indicating BBMap's raw score, prefixed by YR:i:\n",
      "timetag=f               Write a tag indicating this read's mapping time, prefixed by X0:i:\n",
      "boundstag=f             Write a tag indicating whether either read in the pair\n",
      "                        goes off the end of the reference, prefixed by XB:Z:\n",
      "notags=f                Turn off all optional tags.\n",
      "\n",
      "Histogram and statistics output parameters:\n",
      "\n",
      "scafstats=<file>        Statistics on how many reads mapped to which scaffold.\n",
      "refstats=<file>         Statistics on how many reads mapped to which reference\n",
      "                        file; only for BBSplit.\n",
      "sortscafs=t             Sort scaffolds or references by read count.\n",
      "bhist=<file>            Base composition histogram by position.\n",
      "qhist=<file>            Quality histogram by position.\n",
      "aqhist=<file>           Histogram of average read quality.\n",
      "bqhist=<file>           Quality histogram designed for box plots.\n",
      "lhist=<file>            Read length histogram.\n",
      "ihist=<file>            Write histogram of insert sizes (for paired reads).\n",
      "ehist=<file>            Errors-per-read histogram.\n",
      "qahist=<file>           Quality accuracy histogram of error rates versus \n",
      "                        quality score.\n",
      "indelhist=<file>        Indel length histogram.\n",
      "mhist=<file>            Histogram of match, sub, del, and ins rates by \n",
      "                        read location.\n",
      "gchist=<file>           Read GC content histogram.\n",
      "gcbins=100              Number gchist bins.  Set to 'auto' to use read length.\n",
      "idhist=<file>           Histogram of read count versus percent identity.\n",
      "idbins=100              Number idhist bins.  Set to 'auto' to use read length.\n",
      "statsfile=stderr        Mapping statistics are printed here.\n",
      "\n",
      "Coverage output parameters (these may reduce speed and use more RAM):\n",
      "\n",
      "covstats=<file>         Per-scaffold coverage info.\n",
      "rpkm=<file>             Per-scaffold RPKM/FPKM counts.\n",
      "covhist=<file>          Histogram of # occurrences of each depth level.\n",
      "basecov=<file>          Coverage per base location.\n",
      "bincov=<file>           Print binned coverage per location (one line per X bases).\n",
      "covbinsize=1000         Set the binsize for binned coverage output.\n",
      "nzo=t                   Only print scaffolds with nonzero coverage.\n",
      "twocolumn=f             Change to true to print only ID and Avg_fold instead of \n",
      "                        all 6 columns to the 'out=' file.\n",
      "32bit=f                 Set to true if you need per-base coverage over 64k.\n",
      "strandedcov=f           Track coverage for plus and minus strand independently.\n",
      "startcov=f              Only track start positions of reads.\n",
      "secondarycov=t          Include coverage of secondary alignments.\n",
      "physcov=f               Calculate physical coverage for paired reads.\n",
      "                        This includes the unsequenced bases.\n",
      "\n",
      "Java Parameters:\n",
      "-Xmx                    This will be passed to Java to set memory usage, \n",
      "                        overriding the program's automatic memory detection.\n",
      "                        -Xmx20g will specify 20 gigs of RAM, and -Xmx800m \n",
      "                        will specify 800 megs.  The max is typically 85% of \n",
      "                        physical memory.  The human genome requires around 24g,\n",
      "                        or 12g with the 'usemodulo' flag.  The index uses \n",
      "                        roughly 6 bytes per reference base.\n",
      "\n",
      "This list is not complete.  For more information, please consult \n",
      "/nobackup1/jbrown/bbmap/docs/readme.txt. \n",
      "Please contact Brian Bushnell at bbushnell@lbl.gov if you encounter \n",
      "any problems, or post at: http://seqanswers.com/forums/showthread.php?t=41057\n"
     ]
    },
    {
     "cell_type": "raw",
     "metadata": {},
     "source": [
      "#General input for assembly:\n",
      "tadpole.sh in=<reads> out=<contigs>\n",
      "\n",
      "#mapping with bbmap\n",
      "bbmap.sh ref=<reference fasta> in=<reads> out=<output sam> nodisk\n",
      "\n",
      "#sort and map coverage:\n",
      "samtools sort \"+name+\"step1.contig1.bam \"+name+\"sorted.step1.contig1\\nsamtools index \"+name+\"sorted.step1.contig1.bam\\n\\nbedtools genomecov -ibam \"+name+\"sorted.step1.contig1.bam -d > \"+name+\"step1.contig1.genomecoverage.txt\\n\\n\")\n",
      "\n",
      "#going to start with a default run with normal kmer length etc.  \n",
      "#files to use will be:\n",
      "From group 4: 1.270.A. and 1.272.O.\n",
      "From group 5: 1.245.O. and 1.150.O.\n",
      "From group 15: 1.011.O. and 1.048.O.\n",
      "\n",
      "map_back=\"bash bbmap.sh ref=\"+phage+\"tadpole.fasta in=../\"+phage+\"*fastq out=\"phage+\"tadpole.sam\\n\"\n",
      "same2bam=\"samtools view -bS \"+phage+\"tadpole.sam > \"+phage+\"tadpole.bam\\n\"\n",
      "genomecov=\"samtools sort \"+phage+\"tadpole.bam \"+phage+\"sorted.tadpole.bam\""
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import os\n",
      "os.mkdir(\"assemble_test\")\n",
      "os.chdir(\"assemble_test\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "testlist=[\"1.270.A.\", \"1.272.O.\",\"1.245.O.\", \"1.150.O.\", \"1.011.O.\", \"1.048.O.\"]\n",
      "fastqs=[\"../1.272.O._10N.286.54.C4.fastq\",\"../1.245.O._10N.261.54.C7.fastq\",\"../1.150.O._10N.222.46.A6.fastq\",\"../1.011.O._10N.286.49.B11.fastq\",\"../1.048.O._10N.286.46.A10.fastq\",\"../1.270.A._10N.286.54.A8.fastq\"]\n",
      "\n",
      "out=open(\"tadpole.slurm\",\"w\")\n",
      "out.write(\"#!/bin/bash\\n#SBATCH -n 16\\n#SBATCH -N 1\\n#SBATCH -o s1c1.out\\n#SBATCH -e s1c1.err\\n#SBATCH -p sched_mit_chisholm\\n\\n\")\n",
      "#out.write(\"module add engaging/clc-assembly-cell\\nmodule add engaging/samtools/1.1\\nmodule add engaging/bedtools/2.22.1\\n\\n\")\n",
      "\n",
      "for i in range(len(testlist)):\n",
      "    tadpole=\"bash /nobackup1/jbrown/bbmap/tadpole.sh in=\"+fastqs[i]+\" out=\"+testlist[i]+\"tadpole.fasta\\n\"\n",
      "    out.write(tadpole)\n",
      "out.close()\n",
      "!open tadpole.slurm -a textedit\n",
      "#!scp tadpole.slurm jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/assemble_test/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 12
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      ">After running the above script on the server, tadpole was unable to use some of the fastq files due to a strange error.  Other assemblers have not had issues with these files."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#now run spades:"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#Try to run spades on assemblies to see if this makes a difference in the repeat regions at the ends of the contigs:\n",
      "out=open(\"spades.slurm\",\"w\")\n",
      "out.write(\"#!/bin/bash\\n\")\n",
      "out.write(\"#SBATCH -n 16\\n#SBATCH -N 2\\n#SBATCH -o spades.out\\n#SBATCH -e spades.err \\n#SBATCH -p sched_mit_chisholm\\n\")\n",
      "out.write(\"#SBATCH --mem 60000\\n#SBATCH --exclusive\\n\\n\")\n",
      "out.write(\"module load engaging/SPAdes/3.1.1\\n\")\n",
      "out.write(\"mkdir spades_run1\\n\\n\")\n",
      "out1=\"./spades_run1/\"\n",
      "\n",
      "for virus in testlist:\n",
      "    reads=\"../\"+virus+\"*.fastq\"\n",
      "    outputdir=out1+virus+\"/\"\n",
      "    out.write(\"spades.py -o \"+outputdir+\" --12 \"+reads+\"\\n\")\n",
      "\n",
      "out.close()\n",
      "\n",
      "!open spades.slurm -a textedit\n",
      "!scp spades.slurm jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/assemble_test/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "spades.slurm                                    0%    0     0.0KB/s   --:-- ETA\r",
        "spades.slurm                                  100%  574     0.6KB/s   00:00    \r\n"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for line in testlist:\n",
      "    print(\"cp ./\"+line+\"/contigs.fasta ./\"+line+\"spades1.fasta\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cp ./1.270.A./contigs.fasta ./1.270.A.spades1.fasta\n",
        "cp ./1.272.O./contigs.fasta ./1.272.O.spades1.fasta\n",
        "cp ./1.245.O./contigs.fasta ./1.245.O.spades1.fasta\n",
        "cp ./1.150.O./contigs.fasta ./1.150.O.spades1.fasta\n",
        "cp ./1.011.O./contigs.fasta ./1.011.O.spades1.fasta\n",
        "cp ./1.048.O./contigs.fasta ./1.048.O.spades1.fasta\n"
       ]
      }
     ],
     "prompt_number": 18
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!scp jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/assemble_test/*spades1.fasta ./"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.011.O.spades1.fasta                           0%    0     0.0KB/s   --:-- ETA"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.011.O.spades1.fasta                         100%   36KB  36.3KB/s   00:00    \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.048.O.spades1.fasta                           0%    0     0.0KB/s   --:-- ETA"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.048.O.spades1.fasta                         100%  534KB 533.5KB/s   00:00    \r\n",
        "\r",
        "1.150.O.spades1.fasta                           0%    0     0.0KB/s   --:-- ETA"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.150.O.spades1.fasta                         100%   76KB  75.9KB/s   00:00    \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.245.O.spades1.fasta                           0%    0     0.0KB/s   --:-- ETA\r",
        "1.245.O.spades1.fasta                         100%   75KB  74.8KB/s   00:00    \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.270.A.spades1.fasta                           0%    0     0.0KB/s   --:-- ETA\r",
        "1.270.A.spades1.fasta                         100%   61KB  60.9KB/s   00:00    \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r",
        "1.272.O.spades1.fasta                           0%    0     0.0KB/s   --:-- ETA\r",
        "1.272.O.spades1.fasta                         100%  102KB 101.9KB/s   00:00    \r\n"
       ]
      }
     ],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from pyfaidx import Fasta\n",
      "\n",
      "for line in testlist:\n",
      "    f=Fasta(line+\"spades1.fasta\")\n",
      "    out=open(line+\"longest_contig.fasta\",\"w\")\n",
      "    maxlen=0\n",
      "    for entry in f.keys():\n",
      "        seqlen=len(str(f[entry]))\n",
      "        if seqlen>maxlen:\n",
      "            maxlen=seqlen\n",
      "            title=entry\n",
      "    sequence=str(f[title])\n",
      "    \n",
      "    length=title.split(\"_\")[3]\n",
      "    \n",
      "    coverage=title.split(\"_\")[5]\n",
      "    out.write(\">\"+line+\"\\n\"+sequence+\"\\n\")\n",
      "    out.close()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 36
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls *longest_contig.fasta"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1.011.O.longest_contig.fasta  1.245.O.longest_contig.fasta\r\n",
        "1.048.O.longest_contig.fasta  1.270.A.longest_contig.fasta\r\n",
        "1.150.O.longest_contig.fasta  1.272.O.longest_contig.fasta\r\n"
       ]
      }
     ],
     "prompt_number": 28
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "####checking out the longest contig compared to the final contigs made using CLC as the assembler, it appears that the two assembly methods (SPADES and CLC) assembled very similar contigs that had similar assembly problems of inability to id the start location for contigs that had terminal repeats or that were headful phages.####\n",
      ">synteny was very similar fot those two groups.  Some cuts in synteny for the group15 phage 1.011.O. between assemblies."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}