{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import os.path as op\n",
    "if op.exists(\"data\") == False:\n",
    "    os.mkdir(\"data\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r",
      "phages.txt                                      0%    0     0.0KB/s   --:-- ETA\r",
      "phages.txt                                    100% 2583     2.5KB/s   00:00    \r\n"
     ]
    }
   ],
   "source": [
    "!scp jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/phages.txt data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(phagelist) as ih, open(\"multicontig.txt\", \"w\") as oh1, open(\"asm_contig_cts.txt\", \"w\") as oh2:\n",
    "    print(\"phage\", \"assembly1_contig_count\", sep=\"\\t\", file=oh2)\n",
    "    for p in phagelist:\n",
    "        asm1 = \"/nobackup1/jbrown/nahant_phage_reads/big_pipeline/{p}step1.contig1.fasta\"\n",
    "        contigs = 0\n",
    "        with open(asm1) as asm:\n",
    "            for l in asm1:\n",
    "                if \">\" in l:\n",
    "                    contigs += 1\n",
    "\n",
    "        if contigs > 1:\n",
    "            print(p, file = oh1)\n",
    "        print(p, count, sep=\"\\t\", file=oh2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "os.mkdir(\"scripts\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.003.O.\n",
      "1.004.O.\n",
      "1.005.O.\n",
      "1.007.O.\n",
      "1.008.O.\n",
      "1.009.O.\n",
      "1.011.O.\n",
      "1.012.O.\n",
      "1.013.O.\n",
      "1.015.O.\n",
      "assembly1_contigs.py                          100%  925     0.9KB/s   00:00    \n"
     ]
    }
   ],
   "source": [
    "!head ./data/phages.txt\n",
    "!scp ./scripts/assembly1_contigs.py jbrown@eofe4.mit.edu:~/scripts/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Want to figure out how many of the initial assemblies have more than one contig associated with it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(phagelist) as ih, open(\"multicontig.txt\", \"w\") as oh1, open(\"asm_contig_cts.txt\", \"w\") as oh2:\n",
    "    print(\"phage\", \"assembly1_contig_count\", sep=\"\\t\", file=oh2)\n",
    "    for phage in ih:\n",
    "        p = phage.strip()\n",
    "        asm1 = \"/nobackup1/jbrown/nahant_phage_reads/big_pipeline/{p}step1.clc.fasta\".format(p=p)\n",
    "        contigs = 0\n",
    "        if op.exists(asm1):\n",
    "            with open(asm1) as asm:\n",
    "                for l in asm:\n",
    "                    if \">\" in l:\n",
    "                        print(l)\n",
    "                        contigs += 1\n",
    "\n",
    "            if contigs > 1:\n",
    "                print(p, file = oh1)\n",
    "\n",
    "            print(p, contigs, sep=\"\\t\", file=oh2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "multicontig.txt                               100% 1179     1.2KB/s   00:00    \n"
     ]
    }
   ],
   "source": [
    "!scp jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/multicontig.txt ./data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "asm_contig_cts.txt                            100% 3227     3.2KB/s   00:00    \n"
     ]
    }
   ],
   "source": [
    "!scp jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/asm_contig_cts.txt ./data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "beginning = '''#!/bin/bash\n",
    "#SBATCH -n 16\n",
    "#SBATCH -N 1\n",
    "#SBATCH -o check_asm1_bedcov.out\n",
    "#SBATCH -e check_asm_bedcov.err\n",
    "#SBATCH -p sched_mit_chisholm\n",
    "\n",
    "module add engaging/clc-assembly-cell\n",
    "module add engaging/samtools/1.1\n",
    "module add engaging/bedtools/2.22.1\n",
    "'''\n",
    "\n",
    "phagelist = \"./data/multicontig.txt\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write slurm submission for read recruitment on multi-contig assemblies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "IOError",
     "evalue": "[Errno 2] No such file or directory: 'phages.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIOError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-a489ce441eb7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0m__future__\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mprint_function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mphagelist\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mih\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"final_coverage.slurm\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"w\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0moh\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbeginning\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0moh\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mp\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mih\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIOError\u001b[0m: [Errno 2] No such file or directory: 'phages.txt'"
     ]
    }
   ],
   "source": [
    "from __future__ import print_function\n",
    "\n",
    "with open(\"./data/multicontig.txt\") as ih, open(\"multicheck.slurm\",\"w\") as oh:\n",
    "    print(beginning, file=oh)\n",
    "    for p in ih:\n",
    "        phage=p.strip()\n",
    "\n",
    "        pdir = \"/nobackup1/jbrown/nahant_phage_reads/\"\n",
    "        wdir = \"multi_contig_check\"\n",
    "        cdir = \"big_pipeline\"\n",
    "        fastq = op.join(pdir,\"{phage}_*.fastq\".format(**locals()))\n",
    "        contig_assembly = op.join(pdir, cdir, \"{phage}step1.clc.fasta\".format(**locals()))\n",
    "        outcas = op.join(pdir, wdir, \"{phage}step1.clc.cas\".format(**locals()))\n",
    "        outbam = op.join(pdir, wdir, \"{phage}step1.clc.bam\".format(**locals()))\n",
    "        rearranged = op.join(pdir, wdir, \"{phage}step1.clc.sorted\".format(**locals()))\n",
    "        genomecov = op.join(pdir, wdir, \"{phage}step1.clc.genomecoverage\".format(**locals()))\n",
    "\n",
    "        clc_map_cmd = \"clc_mapper -o {outcas} -p fb ss 100 300 -q {fastq} -d {contig_assembly} -a local -r ignore\".format(**locals())\n",
    "        clc_to_sam = \"clc_cas_to_sam -a {outcas} -o {outbam}\".format(**locals())\n",
    "        sam_sort = \"samtools sort {outbam} {rearranged}\".format(**locals())\n",
    "        sam_index = \"samtools index {rearranged}\".format(**locals())\n",
    "        bed_gcov = \"bedtools genomecov -ibam {rearranged} -d > {genomecov}\".format(**locals())\n",
    "        \n",
    "        print(clc_map_cmd, clc_to_sam, sam_sort, sam_index, bed_gcov, sep=\"\\n\", file=oh)\n",
    "        print(\"\\n\", file=oh)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!open multicheck.slurm -a textedit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r",
      "multicheck.slurm                                0%    0     0.0KB/s   --:-- ETA\r",
      "multicheck.slurm                              100%  117KB 116.7KB/s   00:00    \r\n"
     ]
    }
   ],
   "source": [
    "!scp multicheck.slurm jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "from __future__ import print_function\n",
    "import os.path as op\n",
    "\n",
    "with open(\"./data/multicontig.txt\") as ih, open(\"multibed.slurm\",\"w\") as oh:\n",
    "    print(beginning, file=oh)\n",
    "    for p in ih:\n",
    "        phage=p.strip()\n",
    "        pdir = \"/nobackup1/jbrown/nahant_phage_reads/\"\n",
    "        wdir = \"multi_contig_check\"\n",
    "        cdir = \"big_pipeline\"\n",
    "        rearranged = op.join(pdir, wdir, \"{phage}step1.clc.sorted.bam\".format(**locals()))\n",
    "        genomecov = op.join(pdir, wdir, \"{phage}step1.clc.genomecoverage\".format(**locals()))\n",
    "        \n",
    "        bed_gcov = \"bedtools genomecov -ibam {rearranged} -d > {genomecov}\".format(**locals())\n",
    "        \n",
    "        print(bed_gcov, file=oh)\n",
    "        print(\"\\n\", file=oh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!open multibed.slurm -a textedit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r",
      "multibed.slurm                                  0%    0     0.0KB/s   --:-- ETA\r",
      "multibed.slurm                                100%   27KB  26.6KB/s   00:00    \r\n"
     ]
    }
   ],
   "source": [
    "!scp multibed.slurm jbrown@eofe4.mit.edu:/nobackup1/jbrown/nahant_phage_reads/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phage\tassembly1_contig_count\n",
      "\n",
      "1.003.O.\t1\n",
      "\n",
      "1.004.O.\t61\n",
      "\n",
      "1.005.O.\t67\n",
      "\n",
      "1.007.O.\t27\n",
      "\n",
      "1.008.O.\t1\n",
      "\n",
      "1.009.O.\t44\n",
      "\n",
      "1.011.O.\t5\n",
      "\n",
      "1.012.O.\t26\n",
      "\n",
      "1.013.O.\t93\n",
      "\n",
      "1.015.O.\t1\n",
      "\n",
      "1.016.O.\t62\n",
      "\n",
      "1.017.O.\t36\n",
      "\n",
      "1.020.O.\t223\n",
      "\n",
      "1.021.A.\t7\n",
      "\n",
      "1.021.B.\t6\n",
      "\n",
      "1.021.C.\t5\n",
      "\n",
      "1.022.O.\t2\n",
      "\n",
      "1.023.O.\t5\n",
      "\n",
      "1.024.O.\t80\n",
      "\n",
      "1.025.O.\t1\n",
      "\n",
      "1.026.O.\t1\n",
      "\n",
      "1.027.O.\t1\n",
      "\n",
      "1.028.O.\t98\n",
      "\n",
      "1.029.O.\t43\n",
      "\n",
      "1.030.O.\t1\n",
      "\n",
      "1.031.O.\t85\n",
      "\n",
      "1.032.O.\t236\n",
      "\n",
      "1.033.O.\t21\n",
      "\n",
      "1.034.O.\t1\n",
      "\n",
      "1.034.X.\t27\n",
      "\n",
      "1.036.O.\t15\n",
      "\n",
      "1.037.O.\t6\n",
      "\n",
      "1.038.O.\t1\n",
      "\n",
      "1.039.O.\t42\n",
      "\n",
      "1.040.O.\t32\n",
      "\n",
      "1.042.O.\t2\n",
      "\n",
      "1.043.O.\t57\n",
      "\n",
      "1.044.O.\t2\n",
      "\n",
      "1.046.O.\t4\n",
      "\n",
      "1.047.O.\t19\n",
      "\n",
      "1.048.O.\t8\n",
      "\n",
      "1.049.O.\t1\n",
      "\n",
      "1.050.O.\t14\n",
      "\n",
      "1.052.A.\t40\n",
      "\n",
      "1.054.O.\t1\n",
      "\n",
      "1.055.O.\t1\n",
      "\n",
      "1.056.O.\t6\n",
      "\n",
      "1.057.O.\t1\n",
      "\n",
      "1.060.A.\t1\n",
      "\n",
      "1.061.O.\t3\n",
      "\n",
      "1.062.O.\t19\n",
      "\n",
      "1.063.O.\t12\n",
      "\n",
      "1.064.O.\t1\n",
      "\n",
      "1.066.O.\t3\n",
      "\n",
      "1.067.O.\t57\n",
      "\n",
      "1.068.O.\t34\n",
      "\n",
      "1.069.O.\t7\n",
      "\n",
      "1.070.O.\t1\n",
      "\n",
      "1.071.A.\t1\n",
      "\n",
      "1.072.O.\t11\n",
      "\n",
      "1.074.O.\t1\n",
      "\n",
      "1.075.O.\t4\n",
      "\n",
      "1.076.O.\t1\n",
      "\n",
      "1.077.O.\t6\n",
      "\n",
      "1.079.O.\t1\n",
      "\n",
      "1.080.O.\t1\n",
      "\n",
      "1.081.O.\t1\n",
      "\n",
      "1.082.O.\t1\n",
      "\n",
      "1.083.O.\t1\n",
      "\n",
      "1.084.O.\t1\n",
      "\n",
      "1.085.O.\t1\n",
      "\n",
      "1.086.O.\t12\n",
      "\n",
      "1.087.A.\t3\n",
      "\n",
      "1.088.O.\t10\n",
      "\n",
      "1.089.O.\t1\n",
      "\n",
      "1.090.B.\t1\n",
      "\n",
      "1.091.O.\t1\n",
      "\n",
      "1.093.O.\t1\n",
      "\n",
      "1.094.O.\t9\n",
      "\n",
      "1.095.O.\t2\n",
      "\n",
      "1.097.O.\t4\n",
      "\n",
      "1.098.O.\t1\n",
      "\n",
      "1.100.O.\t1\n",
      "\n",
      "1.101.O.\t1\n",
      "\n",
      "1.102.O.\t3\n",
      "\n",
      "1.103.O.\t3\n",
      "\n",
      "1.104.O.\t1\n",
      "\n",
      "1.105.O.\t3\n",
      "\n",
      "1.106.O.\t1\n",
      "\n",
      "1.107.A.\t11\n",
      "\n",
      "1.107.B.\t5\n",
      "\n",
      "1.107.C.\t11\n",
      "\n",
      "1.108.O.\t23\n",
      "\n",
      "1.110.O.\t1\n",
      "\n",
      "1.111.A.\t1\n",
      "\n",
      "1.111.B.\t1\n",
      "\n",
      "1.112.O.\t1\n",
      "\n",
      "1.113.A.\t2\n",
      "\n",
      "1.115.A.\t2\n",
      "\n",
      "1.115.B.\t2\n",
      "\n",
      "1.116.O.\t12\n",
      "\n",
      "1.117.O.\t1\n",
      "\n",
      "1.118.A.\t4\n",
      "\n",
      "1.118.B.\t3\n",
      "\n",
      "1.119.O.\t1\n",
      "\n",
      "1.121.O.\t1\n",
      "\n",
      "1.122.A.\t1\n",
      "\n",
      "1.122.B.\t29\n",
      "\n",
      "1.123.O.\t1\n",
      "\n",
      "1.124.O.\t1\n",
      "\n",
      "1.125.O.\t1\n",
      "\n",
      "1.126.O.\t20\n",
      "\n",
      "1.127.O.\t1\n",
      "\n",
      "1.131.O.\t1\n",
      "\n",
      "1.132.O.\t2\n",
      "\n",
      "1.133.O.\t1\n",
      "\n",
      "1.134.O.\t1\n",
      "\n",
      "1.135.O.\t1\n",
      "\n",
      "1.136.O.\t4\n",
      "\n",
      "1.137.O.\t7\n",
      "\n",
      "1.138.O.\t2\n",
      "\n",
      "1.139.A.\t6\n",
      "\n",
      "1.139.B.\t1\n",
      "\n",
      "1.141.A.\t14\n",
      "\n",
      "1.142.O.\t62\n",
      "\n",
      "1.143.O.\t1\n",
      "\n",
      "1.144.O.\t3\n",
      "\n",
      "1.147.O.\t2\n",
      "\n",
      "1.148.O.\t1\n",
      "\n",
      "1.149.O.\t81\n",
      "\n",
      "1.150.O.\t1\n",
      "\n",
      "1.151.O.\t1\n",
      "\n",
      "1.152.O.\t1\n",
      "\n",
      "1.154.O.\t1\n",
      "\n",
      "1.155.O.\t1\n",
      "\n",
      "1.156.O.\t9\n",
      "\n",
      "1.157.O.\t1\n",
      "\n",
      "1.158.O.\t209\n",
      "\n",
      "1.159.O.\t78\n",
      "\n",
      "1.160.O.\t2\n",
      "\n",
      "1.161.O.\t3\n",
      "\n",
      "1.162.O.\t1\n",
      "\n",
      "1.164.O.\t1\n",
      "\n",
      "1.165.O.\t1\n",
      "\n",
      "1.166.O.\t8\n",
      "\n",
      "1.167.O.\t1\n",
      "\n",
      "1.168.O.\t1\n",
      "\n",
      "1.169.O.\t1\n",
      "\n",
      "1.170.O.\t1\n",
      "\n",
      "1.171.O.\t3\n",
      "\n",
      "1.172.O.\t1\n",
      "\n",
      "1.173.O.\t3\n",
      "\n",
      "1.174.O.\t1\n",
      "\n",
      "1.175.O.\t4\n",
      "\n",
      "1.176.O.\t2\n",
      "\n",
      "1.177.O.\t1\n",
      "\n",
      "1.178.O.\t60\n",
      "\n",
      "1.179.O.\t6\n",
      "\n",
      "1.181.O.\t1\n",
      "\n",
      "1.182.O.\t1\n",
      "\n",
      "1.183.O.\t1\n",
      "\n",
      "1.184.A.\t18\n",
      "\n",
      "1.185.O.\t1\n",
      "\n",
      "1.186.O.\t1\n",
      "\n",
      "1.187.O.\t1\n",
      "\n",
      "1.188.A.\t2\n",
      "\n",
      "1.188.B.\t1\n",
      "\n",
      "1.188.C.\t1\n",
      "\n",
      "1.189.B.\t1\n",
      "\n",
      "1.189.C.\t1\n",
      "\n",
      "1.189.O.\t1\n",
      "\n",
      "1.190.O.\t1\n",
      "\n",
      "1.191.O.\t1\n",
      "\n",
      "1.193.O.\t1\n",
      "\n",
      "1.194.O.\t4\n",
      "\n",
      "1.195.O.\t1\n",
      "\n",
      "1.196.O.\t1\n",
      "\n",
      "1.197.A.\t1\n",
      "\n",
      "1.198.A.\t1\n",
      "\n",
      "1.198.B.\t1\n",
      "\n",
      "1.199.A.\t12\n",
      "\n",
      "1.199.B.\t17\n",
      "\n",
      "1.200.O.\t1\n",
      "\n",
      "1.201.B.\t1\n",
      "\n",
      "1.202.O.\t18\n",
      "\n",
      "1.204.O.\t1\n",
      "\n",
      "1.205.O.\t1\n",
      "\n",
      "1.206.O.\t1\n",
      "\n",
      "1.207.B.\t6\n",
      "\n",
      "1.208.B.\t2\n",
      "\n",
      "1.209.O.\t6\n",
      "\n",
      "1.210.O.\t1\n",
      "\n",
      "1.211.A.\t1\n",
      "\n",
      "1.211.B.\t1\n",
      "\n",
      "1.213.O.\t3\n",
      "\n",
      "1.214.O.\t1\n",
      "\n",
      "1.215.A.\t5\n",
      "\n",
      "1.215.B.\t21\n",
      "\n",
      "1.216.O.\t1\n",
      "\n",
      "1.217.O.\t8\n",
      "\n",
      "1.219.O.\t45\n",
      "\n",
      "1.223.O.\t1\n",
      "\n",
      "1.224.A.\t1\n",
      "\n",
      "1.225.O.\t1\n",
      "\n",
      "1.226.O.\t1\n",
      "\n",
      "1.228.O.\t1\n",
      "\n",
      "1.231.O.\t1\n",
      "\n",
      "1.232.O.\t5\n",
      "\n",
      "1.233.A.\t1\n",
      "\n",
      "1.233.B.\t1\n",
      "\n",
      "1.235.O.\t49\n",
      "\n",
      "1.236.O.\t2\n",
      "\n",
      "1.237.A.\t13\n",
      "\n",
      "1.237.B.\t1\n",
      "\n",
      "1.238.A.\t1\n",
      "\n",
      "1.238.B.\t7\n",
      "\n",
      "1.239.O.\t1\n",
      "\n",
      "1.240.O.\t6\n",
      "\n",
      "1.242.O.\t1\n",
      "\n",
      "1.243.O.\t1\n",
      "\n",
      "1.244.A.\t1\n",
      "\n",
      "1.245.O.\t1\n",
      "\n",
      "1.246.O.\t2\n",
      "\n",
      "1.247.A.\t1\n",
      "\n",
      "1.247.B.\t1\n",
      "\n",
      "1.248.O.\t2\n",
      "\n",
      "1.249.A.\t5\n",
      "\n",
      "1.249.B.\t4\n",
      "\n",
      "1.250.O.\t24\n",
      "\n",
      "1.251.O.\t107\n",
      "\n",
      "1.253.O.\t6\n",
      "\n",
      "1.254.O.\t2\n",
      "\n",
      "1.255.O.\t10\n",
      "\n",
      "1.256.O.\t3\n",
      "\n",
      "1.257.O.\t13\n",
      "\n",
      "1.259.O.\t1\n",
      "\n",
      "1.261.O.\t1\n",
      "\n",
      "1.262.O.\t1\n",
      "\n",
      "1.263.A.\t1\n",
      "\n",
      "1.263.B.\t1\n",
      "\n",
      "1.264.O.\t1\n",
      "\n",
      "1.265.O.\t1\n",
      "\n",
      "1.266.O.\t1\n",
      "\n",
      "1.267.O.\t1\n",
      "\n",
      "1.268.A.\t1\n",
      "\n",
      "1.268.B.\t1\n",
      "\n",
      "1.269.O.\t2\n",
      "\n",
      "1.270.A.\t1\n",
      "\n",
      "1.270.B.\t7\n",
      "\n",
      "1.271.A.\t1\n",
      "\n",
      "1.271.B.\t1\n",
      "\n",
      "1.272.O.\t1\n",
      "\n",
      "1.273.O.\t1\n",
      "\n",
      "1.274.O.\t1\n",
      "\n",
      "1.275.O.\t1\n",
      "\n",
      "1.276.O.\t1\n",
      "\n",
      "1.277.A.\t1\n",
      "\n",
      "1.277.B.\t1\n",
      "\n",
      "1.278.O.\t1\n",
      "\n",
      "1.280.O.\t1\n",
      "\n",
      "1.281.O.\t2\n",
      "\n",
      "1.282.A.\t1\n",
      "\n",
      "1.283.A.\t1\n",
      "\n",
      "1.283.B.\t1\n",
      "\n",
      "1.283.C.\t24\n",
      "\n",
      "1.284.A.\t2\n",
      "\n",
      "1.285.O.\t3\n",
      "\n",
      "1.286.O.\t1\n",
      "\n",
      "1.287.O.\t1\n",
      "\n",
      "1.289.A.\t3\n",
      "\n",
      "1.291.O.\t1\n",
      "\n",
      "1.293.O.\t1\n",
      "\n",
      "12G01.O.\t5\n",
      "\n",
      "2.044.O.\t1\n",
      "\n",
      "2.058.O.\t1\n",
      "\n",
      "2.092.O.\t1\n",
      "\n",
      "2.095.A.\t1\n",
      "\n",
      "2.095.B.\t5\n",
      "\n",
      "2.096.O.\t103\n",
      "\n",
      "2.117.O.\t1\n",
      "\n",
      "2.130.O.\t1\n",
      "\n",
      "2.159.A.\t69\n",
      "\n",
      "2.159.B.\t63\n",
      "\n",
      "2.275.O.\t1\n",
      "\n",
      "3.058.O.\t1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "with open(\"./data/asm_contig_cts.txt\") as ih:\n",
    "    for l in ih:\n",
    "        print(l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    " "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
